test/e2e/network/dns_common.go

```diff
@@ -310,6 +310,71 @@ func generateDNSServerPod(aRecords map[string]string) *v1.Pod {
 	return pod
 }
 
+func generateCoreDNSServerPod(corednsConfig *v1.ConfigMap) *v1.Pod {
+	return &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "e2e-dns-configmap-dns-server-",
+		},
+		Spec: v1.PodSpec{
+			Volumes: []v1.Volume{
+				{
+					Name: "coredns-config",
+					VolumeSource: v1.VolumeSource{
+						ConfigMap: &v1.ConfigMapVolumeSource{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: corednsConfig.Name,
+							},
+						},
+					},
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Name:  "dns",
+					Image: imageutils.GetE2EImage(imageutils.Dnsutils),
+					Command: []string{
+						"/coredns",
+						"-conf", "/etc/coredns/Corefile",
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "coredns-config",
+							MountPath: "/etc/coredns",
+							ReadOnly:  true,
+						},
+					},
+				},
+			},
+			DNSPolicy: "Default",
+		},
+	}
+}
+
+func generateCoreDNSConfigmap(namespaceName string, aRecords map[string]string) *v1.ConfigMap {
+	corefileData := `. {
+	file /etc/coredns/db.dns
+	log
+}`
+	dnsRecords := ".               IN  SOA dns.lish.org. foo.lish.org. 2015082541 7200 3600 1209600 3600"
```

Comment:
If you use the *hosts* plugin instead of *file*, you shouldn't need this SOA record (which I assume is only a place holder to satisfy the *file* plugin).

Reply:
Done, it works with ``hosts`` too. It seems simpler too.